src/ResourceManager/Sql/Commands.Sql/Replication/Services/AzureSqlDatabaseReplicationCommunicator.cs

```diff
@@ -56,30 +58,54 @@ public AzureSqlDatabaseReplicationCommunicator(IAzureContext context)
             if (context.Subscription != Subscription)
             {
                 Subscription = context.Subscription;
-                SqlClient = null;
+                LegacySqlClient = null;
             }
         }
 
         /// <summary>
-        /// Gets the Azure SQL Database
+        /// Gets the specified Azure SQL Database replication link with Lagecy sdk
         /// </summary>
         public Management.Sql.LegacySdk.Models.ReplicationLink GetLink(string resourceGroupName, string serverName, string databaseName, Guid linkId)
         {
-            return GetCurrentSqlClient().DatabaseReplicationLinks.Get(resourceGroupName, serverName, databaseName, linkId.ToString()).ReplicationLink;
+            return GetLegacySqlClient().DatabaseReplicationLinks.Get(resourceGroupName, serverName, databaseName, linkId.ToString()).ReplicationLink;
```

Comment:
These changes aren't needed. There is no change to replication links API so it does not need to be updated. Please revert any change to Links api.

Reply:
@jaredmoo Reverted related changes in replication links API in the next commit.